test/extended/etcd/helpers/helpers.go

```diff
@@ -73,6 +73,30 @@ func CreateNewMasterMachine(ctx context.Context, t TestingT, machineClient machi
 	return clonedMachine.Name, nil
 }
 
+func FailRandomlyChosenMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (string, error) {
+	machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
+	if err != nil {
+		return "", err
+	}
+	if len(machineList.Items) == 0 {
+		return "", fmt.Errorf("no master machine found")
+	}
+
+	var machineToFail *machinev1beta1.Machine
+	//machineIndexToFail := rand.Intn(len(machineList.Items))
+	machineToFail = &machineList.Items[0]
+	// update victim machine's status
+	machineToFail.Status.NodeRef = nil
```

Comment:
Does NodeRef need to be nil? What did you see when you didn't set this nil?

Reply:
I am trying to make the node `failed/unhealthy` .. Isn't setting this to `nil` needed ?

Reply:
I don't think so, if you simply set the phase to Failed, from a Machine API perspective that's considered to be unhealthy. It might be needed for your logic though, where you are trying to simulate that the Node is gone.

In this case, do you need to somehow break the etcd pod to make sure you have an unhealthy member?

Reply:
Yeah that would be gr8 to make it unhealthy first, then the node is being deleted .. Any suggestion how to make the member unhealthy and the node failed ?